python/setup.py

```diff
@@ -181,8 +181,12 @@ def initialize_options(self):
         self.bundle_cython_cpp = strtobool(
             os.environ.get('PYARROW_BUNDLE_CYTHON_CPP', '0'))
 
-        self.with_parquet_encryption = (self.with_parquet_encryption and
-                                        self.with_parquet)
+        if self.with_parquet_encryption and not self.with_parquet:
+            raise ValueError(
+                "Cannot build Parquet encryption without Parquet. Set PYARROW_WITH_PARQUET=1 "
+                "or use --with-parquet-encryption."
```

Comment:
It seems like this error message is not consistent?
```suggestion
                "Cannot build Parquet encryption without Parquet. Set PYARROW_WITH_PARQUET=1 "
                "or use --with-parquet."
```

Also, I'm not sure about this change. What do you think @jorisvandenbossche ?

Reply:
I think it's a good change (with your correction)? If I am reading this correctly, if you currently specify you want to build the parquet encryption submodule, but didn't enable parquet itself, we _silently_ ignore your `PYARROW_WITH_PARQUET_ENCRYPTION=1`. While I think the intention of the user is clear in that case that they want encryption built, and not getting that might be confusing. So an explicit error saying they also need to enable parquet seems useful.